python/src/scipp/plot/plot2d.py

```diff
@@ -0,0 +1,144 @@
+    """
+
+    sp = SciPlot2d(scipp_obj_dict=scipp_obj_dict,
+                   axes=axes,
+                   masks=masks,
+                   ax=ax,
+                   cax=cax,
+                   figsize=figsize,
+                   aspect=aspect,
+                   cmap=cmap,
+                   log=log,
+                   vmin=vmin,
+                   vmax=vmax,
+                   color=color,
+                   logx=logx or logxy,
+                   logy=logy or logxy,
+                   resolution=resolution)
+
+    if filename is not None:
+        sp.savefig(filename)
+
+    return sp
+
+
+class SciPlot2d(SciPlot):
+    def __init__(self,
+                 scipp_obj_dict=None,
+                 axes=None,
+                 masks=None,
+                 ax=None,
+                 cax=None,
+                 figsize=None,
+                 pax=None,
+                 aspect=None,
+                 cmap=None,
+                 log=None,
+                 vmin=None,
+                 vmax=None,
+                 color=None,
+                 logx=False,
+                 logy=False,
+                 resolution=None):
+
+        super().__init__()
+
+        # The main controller module which contains the slider widgets
+        self.controller = PlotController2d(scipp_obj_dict=scipp_obj_dict,
+                                           axes=axes,
+                                           masks=masks,
+                                           cmap=cmap,
+                                           log=log,
+                                           vmin=vmin,
+                                           vmax=vmax,
+                                           color=color,
+                                           logx=logx,
+                                           logy=logy,
+                                           button_options=['X', 'Y'])
+
+        # The model which takes care of all heavy calculations
+        self.model = PlotModel2d(controller=self.controller,
+                                 scipp_obj_dict=scipp_obj_dict,
+                                 resolution=resolution)
+
+        # The view which will display the 2d image and send pick events back to
+        # the controller
+        self.view = PlotView2d(
+            controller=self.controller,
+            ax=ax,
+            cax=cax,
+            figsize=figsize,
+            aspect=aspect,
+            cmap=self.controller.params["values"][
+                self.controller.name]["cmap"],
+            norm=self.controller.params["values"][
+                self.controller.name]["norm"],
+            title=self.controller.name,
+            cbar=self.controller.params["values"][
+                self.controller.name]["cbar"],
+            unit=self.controller.params["values"][
+                self.controller.name]["unit"],
+            mask_cmap=self.controller.params["masks"][
+                self.controller.name]["cmap"],
```

Comment:
This is an example of the backward logic I am trying to discuss above: We should probably not be accessing implementation details of the controller? I think this also implies duplication: For example both here and for `PlotView1d` above the current logic requires to set `unit=self.controller....`. If we reverse the logic, all views would support setting their unit, so the controller can do that in its `__init__`, reducing number of places for bugs. The same goes for several other parameters/settings.